src/audio/comp_module/module.c

```diff
@@ -0,0 +1,329 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 NXP. All rights reserved.
+//
+// Author: Paul Olaru <paul.olaru@nxp.com>
+
+#include <config.h>
+
+#if CONFIG_COMP_MODULE
+
+#include <stdint.h>
+#include <stddef.h> // what is it used for?
+#include <sof/lock.h> // What is this used for?
+#include <sof/list.h>
+#include <sof/stream.h> // What is this used for?
+#include <sof/audio/component.h>
+#include <sof/ipc.h> // What is this used for?
+#include <sof/audio/module.h>
+
+static struct comp_dev *module_new(struct sof_ipc_comp *comp)
+{
+	struct sof_ipc_comp_process *ipc_process =
+		(struct sof_ipc_comp_process *)comp;
+	//size_t bs = ipc_process->size;
```

Comment:
Please don't leave commented code. Just remove it if it is not needed.

Reply:
Partially addressed. For module_cmd I still need to figure out how to interface with the userspace to decide upon which module to use at some point and on its configuration data.

Reply:
Addressed with last push.